parachains/runtimes/assets/asset-hub-polkadot/tests/tests.rs

```diff
@@ -646,3 +651,131 @@ asset_test_utils::include_create_and_manage_foreign_assets_for_local_consensus_p
+				_ => None,
+			}
+		}),
+		bridging_to_asset_hub_kusama,
+		WeightLimit::Unlimited,
+	)
+}
+
+#[test]
+fn receive_reserve_asset_deposited_ksm_from_asset_hub_kusama_works() {
+	const BLOCK_AUTHOR_ACCOUNT: [u8; 32] = [13; 32];
+	asset_test_utils::test_cases_over_bridge::receive_reserve_asset_deposited_from_different_consensus_works::<
+		Runtime,
+		XcmConfig,
+		LocationToAccountId,
+		ForeignAssetsInstance,
+	>(
+		collator_session_keys().add(collator_session_key(BLOCK_AUTHOR_ACCOUNT)),
+		ExistentialDeposit::get(),
+		AccountId::from([73; 32]),
+		AccountId::from(BLOCK_AUTHOR_ACCOUNT),
+		// receiving DOTs
+		(MultiLocation { parents: 2, interior: X1(GlobalConsensus(Kusama)) }, 1000000000000, 1_000_000_000),
+		bridging_to_asset_hub_kusama,
+		(X1(PalletInstance(53)), GlobalConsensus(Kusama), X1(Parachain(1000)))
+	)
+}
+
+/// Tests expected configuration of isolated `pallet_xcm::Config::XcmReserveTransferFilter`.
+#[test]
+fn xcm_reserve_transfer_filter_works() {
+	// prepare assets
+	let only_native_assets = || vec![MultiAsset::from((DotLocation::get(), 1000))];
+	let only_trust_backed_assets = || {
+		vec![MultiAsset::from((
+			AssetIdForTrustBackedAssetsConvert::convert_back(&12345).unwrap(),
+			2000,
+		))]
+	};
+	let only_sibling_foreign_assets =
+		|| vec![MultiAsset::from((MultiLocation::new(1, X1(Parachain(12345))), 3000))];
+	let only_different_global_consensus_foreign_assets = || {
+		vec![MultiAsset::from((
+			MultiLocation::new(2, X2(GlobalConsensus(Kusama), Parachain(12345))),
+			4000,
+		))]
+	};
+
+	// prepare destinations
+	let relaychain = MultiLocation::parent();
+	let sibling_parachain = MultiLocation::new(1, X1(Parachain(1000)));
+	let different_global_consensus_parachain_other_then_asset_hub_kusama =
+		MultiLocation::new(2, X2(GlobalConsensus(Kusama), Parachain(12345)));
+	let bridged_asset_hub = bridging::AssetHubKusama::get();
+
+	// prepare expected test data sets: (destination, assets, expected_result)
+	let test_data = vec![
+		(relaychain, only_native_assets(), true),
+		(relaychain, only_trust_backed_assets(), true),
+		(relaychain, only_sibling_foreign_assets(), true),
+		(relaychain, only_different_global_consensus_foreign_assets(), true),
+		(sibling_parachain, only_native_assets(), true),
+		(sibling_parachain, only_trust_backed_assets(), true),
+		(sibling_parachain, only_sibling_foreign_assets(), true),
+		(sibling_parachain, only_different_global_consensus_foreign_assets(), true),
+		(
+			different_global_consensus_parachain_other_then_asset_hub_kusama,
+			only_native_assets(),
+			true,
+		),
+		(
+			different_global_consensus_parachain_other_then_asset_hub_kusama,
+			only_trust_backed_assets(),
+			true,
+		),
+		(
+			different_global_consensus_parachain_other_then_asset_hub_kusama,
+			only_sibling_foreign_assets(),
+			true,
+		),
+		(
+			different_global_consensus_parachain_other_then_asset_hub_kusama,
+			only_different_global_consensus_foreign_assets(),
+			true,
+		),
```

Comment:
`false`? I thought it would only allow DOT over the bridge.

Reply:
ah, thank you, this is even better catch,
on AHP if destination would be (different than AHK) e.g. `MultiLocation: {parents: 2, X2(GlobalConsensus(Kusama), Parachain(1002))` then this fillter passes ok and I thought that XcmRouter would fail, but I forgot that default `NetworkExportTable` does not match remote interior but just remote network.

I am going to fix this and also tests results will change, I am on it...

Reply:
ok, fixed https://github.com/paritytech/cumulus/pull/2762/commits/bae9f3091de9625fbb34cc18e147f051d4306306

for local_consensus/sibling destination it passes for everything,
and for remote destination it passes only if we have bridging for that destination and filter with `AllowedReserveTransferAssetsLocations` passes